cores/esp8266/WString.h

```diff
@@ -230,6 +229,13 @@ class String {
         bool operator ==(const __FlashStringHelper *rhs) const {
             return equals(rhs);
         }
+        bool operator ==(std::nullptr_t) const {
+            return length() == 0;
+        }
+        [[deprecated("use nullptr instead of NULL")]]
+        bool operator ==(decltype(NULL)) const {
```

Comment:
just note that the trade-off is *someone will write* `obj == 5;` returning false positive :)
(...and iirc ide default to 'no warnings', which does not help...)

Reply:
I think they refer to some old version of GCC (pre 4.6?) with the `__null` stuff.
And I'd like to find some examples of `#define NULL std::nullptr`, as it seems like something for the system header to have

Reply:
The [only one](https://github.com/unicode-org/icu/blob/242ad4723ed579bd1f317846b2c04d96b05750cd/icu4c/source/common/unicode/utypes.h#L179-L190) I could find that is installed on my workstation and probably rarely used.

Reply:
So we could've had this
https://github.com/gcc-mirror/gcc/blob/1815462a6e53465c404f8a5f6116891492d4b50b/gcc/ginclude/stddef.h#L396

But, newlib overrides it
https://github.com/earlephilhower/newlib-xtensa/blob/ebc967552ce827f21fc579fd8c437037c1b472ab/newlib/libc/include/stdlib.h#L60-L62

Reply:
Also, glibc code requesting it
https://code.woboq.org/userspace/glibc/stdlib/stdlib.h.html#_M/__need_NULL